poc/elligator2_hacspec.py

```diff
@@ -1,18 +1,26 @@
 from curve25519 import *
-from speclib import *
+from hacspec.speclib import *
 
-a25519 = felem(486662)
-b25519 = felem(1)
-u25519 = felem(2)
+a25519 = to_felem(486662)
+b25519 = to_felem(1)
+u25519 = to_felem(2)
 
 def f_25519(x:felem_t) -> felem_t:
     return fadd(fmul(x, fsqr(x)), fadd(fmul(a25519, fsqr(x)), x))
 
-def hash2curve25519(r:felem_t) -> point_t:
-    d = felem(p25519 - fmul(a25519,finv(fadd(1,fmul(u25519,fsqr(r))))))
+
+def map2curve25519(r:felem_t) -> point_t:
```

Comment:
The return type should be `felem_t`.

Reply:
Indeed. I was waiting for the square root function to make it complete. For now, I'll change it to felem_t.